### ISSUE TITLE
Cannot update cluster after changing container runtime from docker to cri-o

### ISSUE THREAD

**Issue by msschl** [kubernetes/kubeadm#2597]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):

```sh
$ kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.12", GitCommit:"4bf2e32bb2b9fdeea19ff7cdc1fb51fb295ec407", GitTreeState:"clean", BuildDate:"2021-10-27T17:11:17Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Kubernetes version** (use `kubectl version`): 

```sh
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:02:01Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```
- **Cloud provider or hardware configuration**: self hosted hardware
- **OS** (e.g. from /etc/os-release): 
```sh
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- **Kernel** (e.g. `uname -a`):
```sh
$ uname -a
Linux lxkubemaster01 5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
- **Others**:
```sh
$ crio -v
INFO[0000] Starting CRI-O, version: 1.20.5, git: 57392384dcccfb3b0edde6a814aef649246680a4(clean)
crio version 1.20.5
Version:       1.20.5
GitCommit:     57392384dcccfb3b0edde6a814aef649246680a4
GitTreeState:  clean
BuildDate:     2021-09-16T20:56:00Z
GoVersion:     go1.15.2
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

## What happened?

After changing from docker to cri-o by
- Installing cri-o
- Removing docker and containerd from the system
- Editing `/var/lib/kubelet/kubeadm-flags.env` and passing the new runtime in:
```sh
$ cat /var/lib/kubelet/kubeadm-flags.env
KUBELET_KUBEADM_ARGS="--network-plugin=cni --pod-infra-container-image=k8s.gcr.io/pause:3.1 --container-runtime=remote --container-runtime-endpoint=unix:///var/run/crio/crio.sock"
```

the cluster runs stable with cri-o.
However, now we are unable to update the kubernetes version from v1.20.5 to the latest v1.20.12.

Here are the following command I run to update the cluster and the output:
```sh
$ apt-mark unhold kubeadm kubernetes-cni && apt-get update && apt-get install -y kubeadm=${K8S_VERSION}-00 kubernetes-cni && apt-mark hold kubeadm kubernetes-cni

$ kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.12", GitCommit:"4bf2e32bb2b9fdeea19ff7cdc1fb51fb295ec407", GitTreeState:"clean", BuildDate:"2021-10-27T17:11:17Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}

$ kubeadm upgrade diff
[upgrade/diff] Reading configuration from the cluster...
[upgrade/diff] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W1102 21:23:51.690994   69930 kubelet.go:200] cannot automatically set CgroupDriver when starting the Kubelet: cannot execute 'docker info -f {{.CgroupDriver}}': executable file not found in $PATH

$ kubeadm upgrade plan
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W1102 21:25:28.500171   71305 kubelet.go:200] cannot automatically set CgroupDriver when starting the Kubelet: cannot execute 'docker info -f {{.CgroupDriver}}': executable file not found in $PATH
[preflight] Running pre-flight checks.
[upgrade] Running cluster health checks
[upgrade] Fetching available versions to upgrade to
[upgrade/versions] Cluster version: v1.20.5
[upgrade/versions] kubeadm version: v1.20.12
I1102 21:25:31.318464   71305 version.go:254] remote version is much newer: v1.22.3; falling back to: stable-1.20
[upgrade/versions] Latest stable version: v1.20.12
[upgrade/versions] Latest stable version: v1.20.12
[upgrade/versions] Latest version in the v1.20 series: v1.20.12
[upgrade/versions] Latest version in the v1.20 series: v1.20.12

Components that must be upgraded manually after you have upgraded the control plane with 'kubeadm upgrade apply':
COMPONENT   CURRENT        AVAILABLE
kubelet     2 x v1.20.12   v1.20.12
            1 x v1.20.5    v1.20.12

Upgrade to the latest version in the v1.20 series:

COMPONENT                 CURRENT    AVAILABLE
kube-apiserver            v1.20.5    v1.20.12
kube-controller-manager   v1.20.5    v1.20.12
kube-scheduler            v1.20.5    v1.20.12
kube-proxy                v1.20.5    v1.20.12
CoreDNS                   1.7.0      1.7.0
etcd                      3.4.13-0   3.4.13-0

You can now apply the upgrade by executing the following command:

        kubeadm upgrade apply v1.20.12

_____________________________________________________________________


The table below shows the current state of component configs as understood by this version of kubeadm.
Configs that have a "yes" mark in the "MANUAL UPGRADE REQUIRED" column require manual config upgrade or
resetting to kubeadm defaults before a successful upgrade can be performed. The version to manually
upgrade to is denoted in the "PREFERRED VERSION" column.

API GROUP                 CURRENT VERSION   PREFERRED VERSION   MANUAL UPGRADE REQUIRED
kubeproxy.config.k8s.io   v1alpha1          v1alpha1            no
kubelet.config.k8s.io     v1beta1           v1beta1             no
_____________________________________________________________________

$ kubeadm upgrade apply v1.20.12 --v=5
I1103 10:52:04.442700  795675 apply.go:112] [upgrade/apply] verifying health of cluster
I1103 10:52:04.442766  795675 apply.go:113] [upgrade/apply] retrieving configuration from cluster
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W1103 10:52:04.469446  795675 kubelet.go:200] cannot automatically set CgroupDriver when starting the Kubelet: cannot execute 'docker info -f {{.CgroupDriver}}': executable file not found in $PATH
I1103 10:52:04.472275  795675 common.go:168] running preflight checks
[preflight] Running pre-flight checks.
I1103 10:52:04.472295  795675 preflight.go:87] validating if there are any unsupported CoreDNS plugins in the Corefile
I1103 10:52:04.478190  795675 preflight.go:113] validating if migration can be done for the current CoreDNS release.
[upgrade] Running cluster health checks
I1103 10:52:04.489190  795675 health.go:158] Creating Job "upgrade-health-check" in the namespace "kube-system"
I1103 10:52:04.497997  795675 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 10:52:05.500112  795675 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 10:52:06.500269  795675 health.go:195] Job "upgrade-health-check" in the namespace "kube-system" completed
I1103 10:52:06.500289  795675 health.go:201] Deleting Job "upgrade-health-check" in the namespace "kube-system"
I1103 10:52:06.508103  795675 apply.go:120] [upgrade/apply] validating requested and actual version
I1103 10:52:06.508159  795675 apply.go:136] [upgrade/version] enforcing version skew policies
[upgrade/version] You have chosen to change the cluster version to "v1.20.12"
[upgrade/versions] Cluster version: v1.20.5
[upgrade/versions] kubeadm version: v1.20.12
[upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
[upgrade/prepull] Pulling images required for setting up a Kubernetes cluster
[upgrade/prepull] This might take a minute or two, depending on the speed of your internet connection
[upgrade/prepull] You can also perform this action in beforehand using 'kubeadm config images pull'
exec: "docker": executable file not found in $PATH
docker is required for container runtime
k8s.io/kubernetes/cmd/kubeadm/app/util/runtime.NewContainerRuntime
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/runtime/runtime.go:72
k8s.io/kubernetes/cmd/kubeadm/app/preflight.RunPullImagesCheck
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/preflight/checks.go:1065
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.runApply
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/apply.go:152
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdApply.func1
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/apply.go:77
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:850
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:958
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:895
k8s.io/kubernetes/cmd/kubeadm/app.Run
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:50
main.main
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
        /usr/local/go/src/runtime/proc.go:204
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1374
```

Running `kubeadm upgrade diff` results in a warning, which regarding the issue #2270 can be ignored and should be fixed (I think).
Running `kubeadm upgrade apply v1.20.12` will fail and the cluster is not updated.

To complete here is my kubeadm-config stored in the cluster:
```sh
$ kubectl -n kube-system get cm kubeadm-config -o yaml
apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      extraArgs:
        audit-log-maxage: "30"
        audit-log-maxbackup: "10"
        audit-log-maxsize: "200"
        audit-log-path: /var/log/kubernetes/audit.log
        audit-policy-file: /etc/kubernetes/policies/audit-policy.yaml
        authorization-mode: Node,RBAC
        enable-admission-plugins: NodeRestriction,PodSecurityPolicy,AlwaysPullImages,ServiceAccount
        encryption-provider-config: /etc/kubernetes/encryption.conf
        feature-gates: AdvancedAuditing=true
        profiling: "false"
        tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256
      extraVolumes:
      - hostPath: /etc/kubernetes/policies
        mountPath: /etc/kubernetes/policies
        name: policies
        pathType: DirectoryOrCreate
        readOnly: true
      - hostPath: /var/log/kubernetes
        mountPath: /var/log/kubernetes
        name: log
        pathType: DirectoryOrCreate
      - hostPath: /etc/kubernetes/encryption.conf
        mountPath: /etc/kubernetes/encryption.conf
        name: encription
        pathType: File
        readOnly: true
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta2
    certificatesDir: /etc/kubernetes/pki
    clusterName: kubernetes
    controllerManager:
      extraArgs:
        profiling: "false"
    dns:
      type: CoreDNS
    etcd:
      local:
        dataDir: /var/lib/etcd
    imageRepository: k8s.gcr.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.20.5
    networking:
      dnsDomain: k8s-cluster.company.xyz
      podSubnet: 10.244.0.0/16
      serviceSubnet: 10.96.0.0/12
    scheduler:
      extraArgs:
        profiling: "false"
  ClusterStatus: |
    apiEndpoints:
      lxkubemaster01:
        advertiseAddress: 172.18.2.91
        bindPort: 6443
    apiVersion: kubeadm.k8s.io/v1beta2
    kind: ClusterStatus
kind: ConfigMap
metadata:
  creationTimestamp: "2020-03-13T18:28:04Z"
  name: kubeadm-config
  namespace: kube-system
  resourceVersion: "93112520"
  uid: 9d6632ff-4fec-4aa5-b09c-55cec3cd1608
```

## What you expected to happen?

An update to work properly...

**Comment by msschl** [comment]
Printing the `init-defaults` by `kubeadm config print init-defaults` reveals that in the `InitConfiguration` the `nodeRegistration.criSocket` is set to `/var/run/dockershim.sock`
```sh
$ kubeadm config print init-defaults
W1102 21:31:21.785246   76571 kubelet.go:200] cannot automatically set CgroupDriver when starting the Kubelet: cannot execute 'docker info -f {{.CgroupDriver}}': executable file not found in $PATH
apiVersion: kubeadm.k8s.io/v1beta2
bootstrapTokens:
- groups:
  - system:bootstrappers:kubeadm:default-node-token
  token: abcdef.0123456789abcdef
  ttl: 24h0m0s
  usages:
  - signing
  - authentication
kind: InitConfiguration
localAPIEndpoint:
  advertiseAddress: 1.2.3.4
  bindPort: 6443
nodeRegistration:
  criSocket: /var/run/dockershim.sock
  name: lxkubemaster01
  taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
---
apiServer:
  timeoutForControlPlane: 4m0s
apiVersion: kubeadm.k8s.io/v1beta2
certificatesDir: /etc/kubernetes/pki
clusterName: kubernetes
controllerManager: {}
dns:
  type: CoreDNS
etcd:
  local:
    dataDir: /var/lib/etcd
imageRepository: k8s.gcr.io
kind: ClusterConfiguration
kubernetesVersion: v1.20.0
networking:
  dnsDomain: cluster.local
  serviceSubnet: 10.96.0.0/12
scheduler: {}
```

Is this perhaps the reason for the update not working? If so, how can I update the `InitConfiguration`?

Also, it is a bit odd that in the `ClusterConfiguration` returned by `kubeadm config print init-defaults` the `networking.dnsDomain` doesn't equals the value set by the `kubeadm-config`

**Comment by neolit123** [comment]
hi, please see this about `init-defaults`:

https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-config/

> Note: The output of the command is meant to serve as an example. You must manually edit the output of this command to adapt to your setup. Remove the fields that you are not certain about and kubeadm will try to default them on runtime by examining the host.

in terms of switching the CR there is no official guide for this yet.
and we have not e2e tests...

> Running kubeadm upgrade apply v1.20.12 will fail and the cluster is not updated.

you are not showing the resulting error there, but note that kubeadm annotates the Node objects in the cluster with the CRI socket of the runtime on that node. this is done because it has to be stored somewhere to allow upgrades and the kubeletconfiguration does not support it. i assume you have to patch all Node objects where crio is now used.

the error could be due to something else too.

**Comment by neolit123** [comment]
>  If so, how can I update the InitConfiguration?

init / join configuration is only relevant during the execution of the relevant commands.
there is no persisting of init / join configuration in k8s resources (unlike ClusterConfiguration which is stored in kubeadm-config)

**Comment by msschl** [comment]
@neolit123 
Oh sorry, I forgot copying the `kubeadm upgrade apply v1.20.12 --v=5` command and the exception...

```sh
$ kubeadm upgrade apply v1.20.12 --v=5
I1103 10:52:04.442700  795675 apply.go:112] [upgrade/apply] verifying health of cluster
I1103 10:52:04.442766  795675 apply.go:113] [upgrade/apply] retrieving configuration from cluster
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W1103 10:52:04.469446  795675 kubelet.go:200] cannot automatically set CgroupDriver when starting the Kubelet: cannot execute 'docker info -f {{.CgroupDriver}}': executable file not found in $PATH
I1103 10:52:04.472275  795675 common.go:168] running preflight checks
[preflight] Running pre-flight checks.
I1103 10:52:04.472295  795675 preflight.go:87] validating if there are any unsupported CoreDNS plugins in the Corefile
I1103 10:52:04.478190  795675 preflight.go:113] validating if migration can be done for the current CoreDNS release.
[upgrade] Running cluster health checks
I1103 10:52:04.489190  795675 health.go:158] Creating Job "upgrade-health-check" in the namespace "kube-system"
I1103 10:52:04.497997  795675 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 10:52:05.500112  795675 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 10:52:06.500269  795675 health.go:195] Job "upgrade-health-check" in the namespace "kube-system" completed
I1103 10:52:06.500289  795675 health.go:201] Deleting Job "upgrade-health-check" in the namespace "kube-system"
I1103 10:52:06.508103  795675 apply.go:120] [upgrade/apply] validating requested and actual version
I1103 10:52:06.508159  795675 apply.go:136] [upgrade/version] enforcing version skew policies
[upgrade/version] You have chosen to change the cluster version to "v1.20.12"
[upgrade/versions] Cluster version: v1.20.5
[upgrade/versions] kubeadm version: v1.20.12
[upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
[upgrade/prepull] Pulling images required for setting up a Kubernetes cluster
[upgrade/prepull] This might take a minute or two, depending on the speed of your internet connection
[upgrade/prepull] You can also perform this action in beforehand using 'kubeadm config images pull'
exec: "docker": executable file not found in $PATH
docker is required for container runtime
k8s.io/kubernetes/cmd/kubeadm/app/util/runtime.NewContainerRuntime
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/runtime/runtime.go:72
k8s.io/kubernetes/cmd/kubeadm/app/preflight.RunPullImagesCheck
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/preflight/checks.go:1065
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.runApply
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/apply.go:152
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdApply.func1
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/apply.go:77
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:850
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:958
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:895
k8s.io/kubernetes/cmd/kubeadm/app.Run
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:50
main.main
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
        /usr/local/go/src/runtime/proc.go:204
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1374
```

**Comment by msschl** [comment]
@neolit123 
So updating the `kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock` annotations to the crio socket should solve the problem?

```sh
$ kubectl get no -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Node
  metadata:
    annotations:
      kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
      node.alpha.kubernetes.io/ttl: "0"
      projectcalico.org/IPv4Address: 172.18.2.91/16
      projectcalico.org/IPv4IPIPTunnelAddr: 10.244.29.64
      volumes.kubernetes.io/controller-managed-attach-detach: "true"
    creationTimestamp: "2020-03-13T18:28:03Z"
    labels:
      beta.kubernetes.io/arch: amd64
      beta.kubernetes.io/os: linux
      kubernetes.io/arch: amd64
      kubernetes.io/hostname: lxkubemaster01
      kubernetes.io/os: linux
      node-role.kubernetes.io/control-plane: ""
      node-role.kubernetes.io/master: ""
    name: lxkubemaster01
    resourceVersion: "153789549"
    uid: cd886a4c-a62f-4fbb-bf15-14a5cdf7d3c4
  spec:
    podCIDR: 10.244.0.0/24
    podCIDRs:
    - 10.244.0.0/24
    taints:
    - effect: NoSchedule
      key: node-role.kubernetes.io/master
  status:
    addresses:
    - address: 172.18.2.91
      type: InternalIP
    - address: lxkubemaster01
      type: Hostname
    allocatable:
      cpu: "4"
      ephemeral-storage: "36972409590"
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 10102432Ki
      pods: "110"
    capacity:
      cpu: "4"
      ephemeral-storage: 40117632Ki
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 10204832Ki
      pods: "110"
    conditions:
    - lastHeartbeatTime: "2021-11-02T17:14:57Z"
      lastTransitionTime: "2021-11-02T17:14:57Z"
      message: Calico is running on this node
      reason: CalicoIsUp
      status: "False"
      type: NetworkUnavailable
    - lastHeartbeatTime: "2021-11-03T09:56:01Z"
      lastTransitionTime: "2020-09-06T22:01:27Z"
      message: kubelet has sufficient memory available
      reason: KubeletHasSufficientMemory
      status: "False"
      type: MemoryPressure
    - lastHeartbeatTime: "2021-11-03T09:56:01Z"
      lastTransitionTime: "2020-09-06T22:01:27Z"
      message: kubelet has no disk pressure
      reason: KubeletHasNoDiskPressure
      status: "False"
      type: DiskPressure
    - lastHeartbeatTime: "2021-11-03T09:56:01Z"
      lastTransitionTime: "2020-09-06T22:01:27Z"
      message: kubelet has sufficient PID available
      reason: KubeletHasSufficientPID
      status: "False"
      type: PIDPressure
    - lastHeartbeatTime: "2021-11-03T09:56:01Z"
      lastTransitionTime: "2021-11-02T19:24:42Z"
      message: kubelet is posting ready status. AppArmor enabled
      reason: KubeletReady
      status: "True"
      type: Ready
    daemonEndpoints:
      kubeletEndpoint:
        Port: 10250
    images: ...
    nodeInfo:
      architecture: amd64
      bootID: 41a1fcb6-2627-42f6-979e-d945857e1b97
      containerRuntimeVersion: cri-o://1.20.5
      kernelVersion: 5.4.0-89-generic
      kubeProxyVersion: v1.20.5
      kubeletVersion: v1.20.5
      machineID: b9e99b37e66aa23fe008b178580f0914
      operatingSystem: linux
      osImage: Ubuntu 20.04.3 LTS
      systemUUID: 421aacaf-e200-6a3b-ba3b-9ba865315c3e
- apiVersion: v1
  kind: Node
  metadata:
    annotations:
      kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
      node.alpha.kubernetes.io/ttl: "0"
      projectcalico.org/IPv4Address: 172.18.2.92/16
      projectcalico.org/IPv4IPIPTunnelAddr: 10.244.186.128
      volumes.kubernetes.io/controller-managed-attach-detach: "true"
    creationTimestamp: "2020-03-13T18:32:31Z"
    labels:
      beta.kubernetes.io/arch: amd64
      beta.kubernetes.io/os: linux
      kubernetes.io/arch: amd64
      kubernetes.io/hostname: lxkubenode01
      kubernetes.io/os: linux
    name: lxkubenode01
    resourceVersion: "153789508"
    uid: 721bd07b-0428-498d-be51-c1607e411be8
  spec:
    podCIDR: 10.244.1.0/24
    podCIDRs:
    - 10.244.1.0/24
  status:
    addresses:
    - address: 172.18.2.92
      type: InternalIP
    - address: lxkubenode01
      type: Hostname
    allocatable:
      cpu: "18"
      ephemeral-storage: "33160023190"
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 82377156Ki
      pods: "110"
    capacity:
      cpu: "18"
      ephemeral-storage: 35980928Ki
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 82479556Ki
      pods: "110"
    conditions:
    - lastHeartbeatTime: "2021-11-01T11:41:29Z"
      lastTransitionTime: "2021-11-01T11:41:29Z"
      message: Calico is running on this node
      reason: CalicoIsUp
      status: "False"
      type: NetworkUnavailable
    - lastHeartbeatTime: "2021-11-03T09:55:41Z"
      lastTransitionTime: "2021-11-01T11:41:16Z"
      message: kubelet has sufficient memory available
      reason: KubeletHasSufficientMemory
      status: "False"
      type: MemoryPressure
    - lastHeartbeatTime: "2021-11-03T09:55:41Z"
      lastTransitionTime: "2021-11-01T11:41:16Z"
      message: kubelet has no disk pressure
      reason: KubeletHasNoDiskPressure
      status: "False"
      type: DiskPressure
    - lastHeartbeatTime: "2021-11-03T09:55:41Z"
      lastTransitionTime: "2021-11-01T11:41:16Z"
      message: kubelet has sufficient PID available
      reason: KubeletHasSufficientPID
      status: "False"
      type: PIDPressure
    - lastHeartbeatTime: "2021-11-03T09:55:41Z"
      lastTransitionTime: "2021-11-01T11:41:16Z"
      message: kubelet is posting ready status. AppArmor enabled
      reason: KubeletReady
      status: "True"
      type: Ready
    daemonEndpoints:
      kubeletEndpoint:
        Port: 10250
    images: ...
    nodeInfo:
      architecture: amd64
      bootID: 6fe0bcaa-0e8d-4b71-b6f8-e5571c80195d
      containerRuntimeVersion: cri-o://1.20.5
      kernelVersion: 5.4.0-89-generic
      kubeProxyVersion: v1.20.12
      kubeletVersion: v1.20.12
      machineID: b9e99b37e66aa23fe008b178580f0914
      operatingSystem: linux
      osImage: Ubuntu 20.04.3 LTS
      systemUUID: 421a1fea-bcc3-6098-6662-6b3f6100ba06
- apiVersion: v1
  kind: Node
  metadata:
    annotations:
      kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
      node.alpha.kubernetes.io/ttl: "0"
      projectcalico.org/IPv4Address: 172.18.2.93/16
      projectcalico.org/IPv4IPIPTunnelAddr: 10.244.206.64
      volumes.kubernetes.io/controller-managed-attach-detach: "true"
    creationTimestamp: "2020-03-13T18:32:38Z"
    labels:
      beta.kubernetes.io/arch: amd64
      beta.kubernetes.io/os: linux
      kubernetes.io/arch: amd64
      kubernetes.io/hostname: lxkubenode02
      kubernetes.io/os: linux
    name: lxkubenode02
    resourceVersion: "153789099"
    uid: 19dfac55-5483-463e-97ce-55702e8a90bf
  spec:
    podCIDR: 10.244.2.0/24
    podCIDRs:
    - 10.244.2.0/24
  status:
    addresses:
    - address: 172.18.2.93
      type: InternalIP
    - address: lxkubenode02
      type: Hostname
    allocatable:
      cpu: "18"
      ephemeral-storage: "28400257797"
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 41089496Ki
      pods: "110"
    capacity:
      cpu: "18"
      ephemeral-storage: 30816252Ki
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 41191896Ki
      pods: "110"
    conditions:
    - lastHeartbeatTime: "2021-11-01T11:50:38Z"
      lastTransitionTime: "2021-11-01T11:50:38Z"
      message: Calico is running on this node
      reason: CalicoIsUp
      status: "False"
      type: NetworkUnavailable
    - lastHeartbeatTime: "2021-11-03T09:52:21Z"
      lastTransitionTime: "2021-11-01T11:50:24Z"
      message: kubelet has sufficient memory available
      reason: KubeletHasSufficientMemory
      status: "False"
      type: MemoryPressure
    - lastHeartbeatTime: "2021-11-03T09:52:21Z"
      lastTransitionTime: "2021-11-01T11:50:24Z"
      message: kubelet has no disk pressure
      reason: KubeletHasNoDiskPressure
      status: "False"
      type: DiskPressure
    - lastHeartbeatTime: "2021-11-03T09:52:21Z"
      lastTransitionTime: "2021-11-01T11:50:24Z"
      message: kubelet has sufficient PID available
      reason: KubeletHasSufficientPID
      status: "False"
      type: PIDPressure
    - lastHeartbeatTime: "2021-11-03T09:52:21Z"
      lastTransitionTime: "2021-11-01T11:50:24Z"
      message: kubelet is posting ready status. AppArmor enabled
      reason: KubeletReady
      status: "True"
      type: Ready
    daemonEndpoints:
      kubeletEndpoint:
        Port: 10250
    images: ...
    nodeInfo:
      architecture: amd64
      bootID: 161cb8eb-c940-4131-882f-1589b289f58b
      containerRuntimeVersion: cri-o://1.20.5
      kernelVersion: 5.4.0-89-generic
      kubeProxyVersion: v1.20.12
      kubeletVersion: v1.20.12
      machineID: b9e99b37e66aa23fe008b178580f0914
      operatingSystem: linux
      osImage: Ubuntu 20.04.3 LTS
      systemUUID: 421ac903-e259-08d5-cbbd-4fdc22994f39
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Comment by msschl** [comment]
@neolit123 
Ok after annotation all node with `` the `kubeadm upgrade apply v1.20.12` works without any errors:

```sh
$ kubectl annotate node lxkubemaster01 --overwrite kubeadm.alpha.kubernetes.io/cri-socket=/var/run/crio/crio.sock
$ kubectl annotate node lxkubenode01 --overwrite kubeadm.alpha.kubernetes.io/cri-socket=/var/run/crio/crio.sock
$ kubectl annotate node lxkubenode02 --overwrite kubeadm.alpha.kubernetes.io/cri-socket=/var/run/crio/crio.sock

$ kubeadm upgrade apply v1.20.12 --v=5
I1103 11:06:17.463879  808445 apply.go:112] [upgrade/apply] verifying health of cluster
I1103 11:06:17.463925  808445 apply.go:113] [upgrade/apply] retrieving configuration from cluster
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
I1103 11:06:17.491438  808445 common.go:168] running preflight checks
[preflight] Running pre-flight checks.
I1103 11:06:17.491474  808445 preflight.go:87] validating if there are any unsupported CoreDNS plugins in the Corefile
I1103 11:06:17.498554  808445 preflight.go:113] validating if migration can be done for the current CoreDNS release.
[upgrade] Running cluster health checks
I1103 11:06:17.503476  808445 health.go:158] Creating Job "upgrade-health-check" in the namespace "kube-system"
I1103 11:06:17.512028  808445 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 11:06:18.514010  808445 health.go:188] Job "upgrade-health-check" in the namespace "kube-system" is not yet complete, retrying
I1103 11:06:19.514166  808445 health.go:195] Job "upgrade-health-check" in the namespace "kube-system" completed
I1103 11:06:19.514185  808445 health.go:201] Deleting Job "upgrade-health-check" in the namespace "kube-system"
I1103 11:06:19.522088  808445 apply.go:120] [upgrade/apply] validating requested and actual version
I1103 11:06:19.522198  808445 apply.go:136] [upgrade/version] enforcing version skew policies
[upgrade/version] You have chosen to change the cluster version to "v1.20.12"
[upgrade/versions] Cluster version: v1.20.5
[upgrade/versions] kubeadm version: v1.20.12
[upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
[upgrade/prepull] Pulling images required for setting up a Kubernetes cluster
[upgrade/prepull] This might take a minute or two, depending on the speed of your internet connection
[upgrade/prepull] You can also perform this action in beforehand using 'kubeadm config images pull'
I1103 11:06:20.439949  808445 checks.go:845] pulling k8s.gcr.io/kube-apiserver:v1.20.12
I1103 11:06:24.002240  808445 checks.go:845] pulling k8s.gcr.io/kube-controller-manager:v1.20.12
I1103 11:06:26.897552  808445 checks.go:845] pulling k8s.gcr.io/kube-scheduler:v1.20.12
I1103 11:06:29.124456  808445 checks.go:845] pulling k8s.gcr.io/kube-proxy:v1.20.12
I1103 11:06:32.419148  808445 checks.go:839] image exists: k8s.gcr.io/pause:3.2
I1103 11:06:32.437937  808445 checks.go:839] image exists: k8s.gcr.io/etcd:3.4.13-0
I1103 11:06:32.454601  808445 checks.go:845] pulling k8s.gcr.io/coredns:1.7.0
I1103 11:06:34.852060  808445 apply.go:162] [upgrade/apply] performing upgrade
[upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.20.12"...
Static pod: kube-apiserver-lxkubemaster01 hash: 9b9c92618539be901690258c3df84c07
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
I1103 11:06:34.863754  808445 etcd.go:177] retrieving etcd endpoints from "kubeadm.kubernetes.io/etcd.advertise-client-urls" annotation in etcd Pods
I1103 11:06:34.866432  808445 etcd.go:101] etcd endpoints read from pods: https://172.18.2.91:2379
I1103 11:06:34.888080  808445 etcd.go:247] etcd endpoints read from etcd: https://172.18.2.91:2379
I1103 11:06:34.888095  808445 etcd.go:119] update etcd endpoints: https://172.18.2.91:2379
[upgrade/etcd] Upgrading to TLS for etcd
Static pod: etcd-lxkubemaster01 hash: a1b07a1d563796dcc6c141ff3cbb25b2
I1103 11:06:37.484588  808445 local.go:74] [etcd] wrote Static Pod manifest for a local etcd member to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests201582721/etcd.yaml"
[upgrade/staticpods] Preparing for "etcd" upgrade
[upgrade/staticpods] Renewing etcd-server certificate
I1103 11:06:37.486417  808445 certs.go:474] validating certificate period for etcd/ca certificate
[upgrade/staticpods] Renewing etcd-peer certificate
[upgrade/staticpods] Renewing etcd-healthcheck-client certificate
[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/etcd.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests-2021-11-03-11-06-34/etcd.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[upgrade/staticpods] This might take a minute or longer depending on the component/version gap (timeout 5m0s)
Static pod: etcd-lxkubemaster01 hash: a1b07a1d563796dcc6c141ff3cbb25b2
Static pod: etcd-lxkubemaster01 hash: a1b07a1d563796dcc6c141ff3cbb25b2
Static pod: etcd-lxkubemaster01 hash: d8d6adf73ce26ae99372bda1add4dcb5
[apiclient] Found 1 Pods for label selector component=etcd
[upgrade/staticpods] Component "etcd" upgraded successfully!
I1103 11:07:20.476843  808445 etcd.go:177] retrieving etcd endpoints from "kubeadm.kubernetes.io/etcd.advertise-client-urls" annotation in etcd Pods
I1103 11:07:20.479679  808445 etcd.go:101] etcd endpoints read from pods: https://172.18.2.91:2379
I1103 11:07:20.485570  808445 etcd.go:247] etcd endpoints read from etcd: https://172.18.2.91:2379
I1103 11:07:20.485582  808445 etcd.go:119] update etcd endpoints: https://172.18.2.91:2379
[upgrade/etcd] Waiting for etcd to become available
I1103 11:07:20.485591  808445 etcd.go:488] [etcd] attempting to see if all cluster endpoints ([https://172.18.2.91:2379]) are available 1/10
[upgrade/staticpods] Writing new Static Pod manifests to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests201582721"
I1103 11:07:20.502199  808445 manifests.go:42] [control-plane] creating static Pod files
I1103 11:07:20.502215  808445 manifests.go:96] [control-plane] getting StaticPodSpecs
I1103 11:07:20.503012  808445 certs.go:474] validating certificate period for CA certificate
I1103 11:07:20.503096  808445 manifests.go:109] [control-plane] adding volume "ca-certs" for component "kube-apiserver"
I1103 11:07:20.503104  808445 manifests.go:109] [control-plane] adding volume "encription" for component "kube-apiserver"
I1103 11:07:20.503108  808445 manifests.go:109] [control-plane] adding volume "etc-ca-certificates" for component "kube-apiserver"
I1103 11:07:20.503114  808445 manifests.go:109] [control-plane] adding volume "etc-pki" for component "kube-apiserver"
I1103 11:07:20.503120  808445 manifests.go:109] [control-plane] adding volume "k8s-certs" for component "kube-apiserver"
I1103 11:07:20.503125  808445 manifests.go:109] [control-plane] adding volume "log" for component "kube-apiserver"
I1103 11:07:20.503128  808445 manifests.go:109] [control-plane] adding volume "policies" for component "kube-apiserver"
I1103 11:07:20.503133  808445 manifests.go:109] [control-plane] adding volume "usr-local-share-ca-certificates" for component "kube-apiserver"
I1103 11:07:20.503136  808445 manifests.go:109] [control-plane] adding volume "usr-share-ca-certificates" for component "kube-apiserver"
I1103 11:07:20.504135  808445 manifests.go:126] [control-plane] wrote static Pod manifest for component "kube-apiserver" to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests201582721/kube-apiserver.yaml"
I1103 11:07:20.504152  808445 manifests.go:109] [control-plane] adding volume "ca-certs" for component "kube-controller-manager"
I1103 11:07:20.504158  808445 manifests.go:109] [control-plane] adding volume "etc-ca-certificates" for component "kube-controller-manager"
I1103 11:07:20.504164  808445 manifests.go:109] [control-plane] adding volume "etc-pki" for component "kube-controller-manager"
I1103 11:07:20.504171  808445 manifests.go:109] [control-plane] adding volume "flexvolume-dir" for component "kube-controller-manager"
I1103 11:07:20.504177  808445 manifests.go:109] [control-plane] adding volume "k8s-certs" for component "kube-controller-manager"
I1103 11:07:20.504183  808445 manifests.go:109] [control-plane] adding volume "kubeconfig" for component "kube-controller-manager"
I1103 11:07:20.504197  808445 manifests.go:109] [control-plane] adding volume "usr-local-share-ca-certificates" for component "kube-controller-manager"
I1103 11:07:20.504202  808445 manifests.go:109] [control-plane] adding volume "usr-share-ca-certificates" for component "kube-controller-manager"
I1103 11:07:20.504853  808445 manifests.go:126] [control-plane] wrote static Pod manifest for component "kube-controller-manager" to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests201582721/kube-controller-manager.yaml"
I1103 11:07:20.504866  808445 manifests.go:109] [control-plane] adding volume "kubeconfig" for component "kube-scheduler"
I1103 11:07:20.505203  808445 manifests.go:126] [control-plane] wrote static Pod manifest for component "kube-scheduler" to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests201582721/kube-scheduler.yaml"
[upgrade/staticpods] Preparing for "kube-apiserver" upgrade
[upgrade/staticpods] Renewing apiserver certificate
I1103 11:07:20.505450  808445 certs.go:474] validating certificate period for ca certificate
[upgrade/staticpods] Renewing apiserver-kubelet-client certificate
[upgrade/staticpods] Renewing front-proxy-client certificate
I1103 11:07:20.726863  808445 certs.go:474] validating certificate period for front-proxy CA certificate
I1103 11:07:20.728693  808445 certs.go:474] validating certificate period for front-proxy-ca certificate
[upgrade/staticpods] Renewing apiserver-etcd-client certificate
[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/kube-apiserver.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests-2021-11-03-11-06-34/kube-apiserver.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[upgrade/staticpods] This might take a minute or longer depending on the component/version gap (timeout 5m0s)
Static pod: kube-apiserver-lxkubemaster01 hash: 9b9c92618539be901690258c3df84c07
Static pod: kube-apiserver-lxkubemaster01 hash: 9b9c92618539be901690258c3df84c07
Static pod: kube-apiserver-lxkubemaster01 hash: e0d6e604191279c1fe24dabef5fd17bc
[apiclient] Found 1 Pods for label selector component=kube-apiserver
[upgrade/staticpods] Component "kube-apiserver" upgraded successfully!
[upgrade/staticpods] Preparing for "kube-controller-manager" upgrade
[upgrade/staticpods] Renewing controller-manager.conf certificate
[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/kube-controller-manager.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests-2021-11-03-11-06-34/kube-controller-manager.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[upgrade/staticpods] This might take a minute or longer depending on the component/version gap (timeout 5m0s)
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: 8b56c39d403865226294b5cb57c9a3a7
Static pod: kube-controller-manager-lxkubemaster01 hash: e323b178b06a00e6e88429f2795bc3cf
[apiclient] Found 1 Pods for label selector component=kube-controller-manager
[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
[upgrade/staticpods] Preparing for "kube-scheduler" upgrade
[upgrade/staticpods] Renewing scheduler.conf certificate
[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests-2021-11-03-11-06-34/kube-scheduler.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[upgrade/staticpods] This might take a minute or longer depending on the component/version gap (timeout 5m0s)
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: d02f5b909b1f2e6826faaea407bfcd34
Static pod: kube-scheduler-lxkubemaster01 hash: b0d8e533c867b45f0ac7a3a5dcc8c2bd
[apiclient] Found 1 Pods for label selector component=kube-scheduler
[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
[upgrade/postupgrade] Applying label node-role.kubernetes.io/control-plane='' to Nodes with label node-role.kubernetes.io/master='' (deprecated)
I1103 11:08:12.545099  808445 apply.go:175] [upgrade/postupgrade] upgrading RBAC rules and addons
[upload-config] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[kubelet] Creating a ConfigMap "kubelet-config-1.20" in namespace kube-system with the configuration for the kubelets in the cluster
[kubelet-start] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
I1103 11:08:12.578152  808445 patchnode.go:30] [patchnode] Uploading the CRI Socket information "/var/run/crio/crio.sock" to the Node API object "lxkubemaster01" as an annotation
[bootstrap-token] configured RBAC rules to allow Node Bootstrap tokens to get nodes
[bootstrap-token] configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
I1103 11:08:13.157322  808445 request.go:591] Throttling request took 61.407117ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?timeout=10s
I1103 11:08:13.357328  808445 request.go:591] Throttling request took 195.58865ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/kubeadm:kubelet-bootstrap?timeout=10s
[bootstrap-token] configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
I1103 11:08:13.557333  808445 request.go:591] Throttling request took 197.589601ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?timeout=10s
I1103 11:08:13.757318  808445 request.go:591] Throttling request took 196.85135ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/kubeadm:node-autoapprove-bootstrap?timeout=10s
[bootstrap-token] configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
I1103 11:08:13.957278  808445 request.go:591] Throttling request took 197.305646ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?timeout=10s
I1103 11:08:14.157320  808445 request.go:591] Throttling request took 196.565192ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/kubeadm:node-autoapprove-certificate-rotation?timeout=10s
I1103 11:08:14.159834  808445 clusterinfo.go:79] creating the RBAC rules for exposing the cluster-info ConfigMap in the kube-public namespace
I1103 11:08:14.357328  808445 request.go:591] Throttling request took 197.411355ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-public/roles?timeout=10s
I1103 11:08:14.557275  808445 request.go:591] Throttling request took 195.67892ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-public/roles/kubeadm:bootstrap-signer-clusterinfo?timeout=10s
I1103 11:08:14.757330  808445 request.go:591] Throttling request took 197.451372ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-public/rolebindings?timeout=10s
I1103 11:08:14.957336  808445 request.go:591] Throttling request took 196.771413ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-public/rolebindings/kubeadm:bootstrap-signer-clusterinfo?timeout=10s
I1103 11:08:15.157324  808445 request.go:591] Throttling request took 185.827094ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterroles?timeout=10s
I1103 11:08:15.357308  808445 request.go:591] Throttling request took 196.986826ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterroles/system:coredns?timeout=10s
I1103 11:08:15.557311  808445 request.go:591] Throttling request took 197.647717ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?timeout=10s
I1103 11:08:15.757313  808445 request.go:591] Throttling request took 196.645992ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/system:coredns?timeout=10s
[addons] Applied essential addon: CoreDNS
I1103 11:08:15.957294  808445 request.go:591] Throttling request took 87.914702ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?timeout=10s
I1103 11:08:16.157321  808445 request.go:591] Throttling request took 197.193809ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/kubeadm:node-proxier?timeout=10s
I1103 11:08:16.357335  808445 request.go:591] Throttling request took 197.447536ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/roles?timeout=10s
I1103 11:08:16.557308  808445 request.go:591] Throttling request took 197.023742ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/roles/kube-proxy?timeout=10s
I1103 11:08:16.757253  808445 request.go:591] Throttling request took 197.220283ms, request: POST:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/rolebindings?timeout=10s
I1103 11:08:16.957316  808445 request.go:591] Throttling request took 196.880652ms, request: PUT:https://172.18.2.91:6443/apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/rolebindings/kube-proxy?timeout=10s
[addons] Applied essential addon: kube-proxy

[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.20.12". Enjoy!

[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets if you haven't already done so.
```

**Comment by neolit123** [comment]
Great.

We are waiting on the kubelet maintainers to write their guide for a proper CR swap and we can link it from kubeadm docs.
